base/linalg/generic.jl

```diff
@@ -795,6 +795,19 @@ function inv(A::AbstractMatrix{T}) where T
     A_ldiv_B!(factorize(convert(AbstractMatrix{S}, A)), eye(S0, checksquare(A)))
 end
 
+function pinv(v::AbstractVector{T}, tol::Real=real(zero(T))) where T
```

Comment:
> A vector has one singular value s, which obvious is the maximum one.

Of course.

I can see that the `length(v) > 1/eps(real(float(one(T))))` condition is a bit weird although it might not be a big practical concern since it would be 32 PB for `Float64`. However, I did some simulations and I'm wondering if using the maximum length is the right thing to do. It looks like it is the minimum that determines the error and using the minimum would fix the bizarre case for vectors.